models/nc_news_models.js

```diff
@@ -20,7 +20,17 @@ const selectArticleByID = (id) => {
     })
 }
 
-const selectArticles = () => {
+const selectArticles = (query) => {
+    if (Object.keys(query).length > 0) {
+        if (query.topic) {
+            const { topic } = query.topic
+            return db.query(`SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.article_id)::INT AS comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id WHERE articles.topic = $1 GROUP BY articles.article_id ORDER BY articles.created_at DESC;`, [topic]).then((result) => {
+                return result.rows
+            })
+        } else {
+            return Promise.reject({status:400, msg:'Bad request'})
+        }
```

Comment:
The way you're currently building this up may start to get quite repetitive.  Consider putting your query string on a variable with 'let' so that you can modify it as required.